core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -4210,6 +4247,131 @@ public boolean isExistingUser(String userName) throws UserStoreException {
+            }
+
+        } else if (index == 0) {
+            userCount = doCountUsers(filter.substring(1));
+            return userCount;
+        }
+
+        try {
+            userCount = doCountUsers(filter);
+        } catch (UserStoreException ex) {
+            handleGetUserCountFailure(ErrorMessages.ERROR_CODE_ERROR_WHILE_GETTING_COUNT_USERS.getCode(),
+                    String.format(ErrorMessages.ERROR_CODE_ERROR_WHILE_GETTING_COUNT_USERS.getMessage(), ex.getMessage()),
+                    null, null);
+            throw ex;
+        }
+
+        return userCount;
+    }
+
+    /**
+     * Count roles in user stores
+     *
+     * @param filter The string to filter out roles
+     * @return countRoles
+     * @throws UserStoreException UserStoreException
+     */
+    public final Long countRoles(String filter) throws UserStoreException {
+
+        if (!isSecureCall.get()) {
+            Class[] argTypes = new Class[]{String.class};
+            Object object = callSecure("countRoles", new Object[]{filter}, argTypes);
+            return (Long) object;
+        }
+
+        int index;
+        Long countRoles;
+        String domain = null;
+        index = filter.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
+
+        // Check whether we have a secondary UserStoreManager setup.
+        if (index > 0) {
+            domain = filter.substring(0, index);
+            filter = filter.substring(index + 1);
+            UserStoreManager SecondaryUserStoreManager = getSecondaryUserStoreManager(domain);
+            if (SecondaryUserStoreManager != null) {
+                // We have a secondary UserStoreManager registered for this domain.
+                if (SecondaryUserStoreManager instanceof AbstractUserStoreManager) {
+                    return ((AbstractUserStoreManager) SecondaryUserStoreManager).doCountRoles(filter);
+                } else {
+                    return null;
+                }
+            }
+        } else if (index == 0) {
+            countRoles = doCountRoles(filter.substring(1));
+            return countRoles;
+        }
+
+        try {
+            if (domain != null) {
+                if (domain.equals("Internal") || domain.equals("Application")) {
+                    filter = domain + "%" + filter;
+                }
+            }
+            countRoles = doCountRoles(filter);
+
+        } catch (UserStoreException ex) {
+            handleGetUserCountFailure(ErrorMessages.ERROR_CODE_ERROR_WHILE_GETTING_ROLES_COUNT.getCode(),
+                    String.format(ErrorMessages.ERROR_CODE_ERROR_WHILE_GETTING_ROLES_COUNT.getMessage(),
+                            ex.getMessage()), null, null);
+            throw ex;
+        }
+
+        return countRoles;
+    }
+
+    /**
+     * Count Claims in user stores
+     *
+     * @return claim count value
+     * @throws UserStoreException
+     */
+    public final Long countUsersClaims(String claimURI, String valueFilter) throws UserStoreException {
```

Comment:
why do we need to count claims?

Reply:
By this we are count users that filtered with claim, Ill change the method  name to meaning full one